src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -41,13 +41,15 @@ public sealed class LegacyPackageReferenceProject : BuildIntegratedNuGetProject
         private string _projectUniqueName;
         private string _projectFullPath;
         private Dictionary<string, ProjectInstalledPackage> _installedPackages = new Dictionary<string, ProjectInstalledPackage>(StringComparer.OrdinalIgnoreCase);
+        private Dictionary<string, PackageReference> _transitivePackages = new Dictionary<string, PackageReference>(StringComparer.OrdinalIgnoreCase);
         private DateTime _lastTimeAssetsModified;
 
         public LegacyPackageReferenceProject(
             IVsProjectAdapter vsProjectAdapter,
             string projectId,
             INuGetProjectServices projectServices,
-            IVsProjectThreadingService threadingService)
+            IVsProjectThreadingService threadingService,
+            NuGetFramework targetFramework = null)
```

Comment:
Let's avoid optional parameter whenever we can: https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#optional-parameters.

Reply:
Changed to use overloading instead.